src/main/webapp/app/exercises/shared/plagiarism/plagiarism-inspector/plagiarism-inspector.component.ts

```diff
@@ -0,0 +1,135 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { HttpResponse } from '@angular/common/http';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subject, Subscription } from 'rxjs';
+import { ModelingExerciseService, ModelingSubmissionComparisonDTO } from 'app/exercises/modeling/manage/modeling-exercise.service';
+import { ModelingExercise } from 'app/entities/modeling-exercise.model';
+import { downloadFile } from 'app/shared/util/download.util';
+import { ExportToCsv } from 'export-to-csv';
+
+@Component({
+    selector: 'jhi-plagiarism-inspector',
+    styleUrls: ['./plagiarism-inspector.component.scss'],
+    templateUrl: './plagiarism-inspector.component.html',
+})
+export class PlagiarismInspectorComponent implements OnDestroy, OnInit {
+    selectedComparisonIndex: number;
+    checkPlagiarismInProgress: boolean;
+    modelingExercise: ModelingExercise;
+    modelingSubmissionComparisons: Array<ModelingSubmissionComparisonDTO>;
+    splitControlSubject: Subject<string> = new Subject<string>();
+
+    private subscription: Subscription;
+
+    constructor(private route: ActivatedRoute, private router: Router, private modelingExerciseService: ModelingExerciseService) {}
+
+    ngOnInit() {
+        this.subscription = this.route.params.subscribe((params) => {
+            this.modelingExerciseService.find(params['exerciseId']).subscribe((response: HttpResponse<ModelingExercise>) => {
+                this.modelingExercise = response.body!;
+            });
+        });
+    }
+
+    ngOnDestroy() {
+        this.subscription.unsubscribe();
```

Comment:
You do not need to unsubscribe to http subscriptions as far as i know. 
https://medium.com/@lukaonik/do-we-need-to-unsubscribe-http-client-in-angular-86d781522b99